raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -549,9 +571,37 @@ public void testFollowerReplication() throws Exception {
 
     @Test
     public void testLeaderAppendSingleMemberQuorum() throws IOException {
+        leaderAppendSingleMemberQuorum();
+    }
+
+    @Test
+    public void testLeaderAppendSingleMemberQuorumWithFileBasedStore() throws IOException {
+        File testStateFile = TestUtils.tempFile();
+        quorumStateStore = new FileBasedStateStore(testStateFile);
```

Comment:
Can you clarify why this test case requires the `FileBasedStateStore`?

Reply:
This test is trying to walk through the basic single member quorum with file based state store integrated. `leaderAppendSingleMemberQuorum` as a sub module is being called in the middle, and we want to perform extra validation on the generated test file after the inner part is complete.

Reply:
Sure, just trying to understand is what about the `FileBasedStore` makes this an interesting case from the perspective of `KafkaRaftClient`.